share/pot/mount-out.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/sh
+# shellcheck disable=SC3033,SC3040,SC3043
+:
+
+mount-out-help()
+{
+	echo "pot mount-out [-hvwr] -p pot -m mnt"
+	echo '  -h print this help'
+	echo '  -v verbose'
+	echo '  -p pot : the working pot'
+	echo '  -m mnt : the mount point inside the pot'
+}
+
+# $1 pot
+# $2 mount point
+_is_mountpoint_used()
+{
+	local _pname _mnt_p _proot
+	_pname="$1"
+	_mnt_p="${2#/}"
+	_conf=$POT_FS_ROOT/jails/$_pname/conf/fscomp.conf
+	_proot=$POT_FS_ROOT/jails/$_pname/m
+	if grep -q " $_proot/$_mnt_p$" "$_conf" ||
+		grep -q " $_proot/$_mnt_p " "$_conf" ; then
+		# mount point already used
+		return 0 # true
+	fi
+	if grep -q "$_proot/$_mnt_p " "$_conf" ; then
```

Comment:
Maybe explicitly grepping for begin of line (like "^blabla") would be useful here? Also, what about mountpoints containing spaces?

Reply:
this string is not at the begin of the line, so ^ cannot be used.
mountpoints with space shouldn't be an issue, the double quote should preserve everything correctly.

Reply:
I though this grep is looking for exactly that ($_proot/$_mnt_p being used as a source, which I though was on the left hand side of fscomp.conf)? Or could there be cases where quotes exist left of the mount source?

My comment regarding spaces was about a situation where a mountpoint named "/opt/pot/jails/test/m/mnt with space" would also be matched by this grep for a mountpoint named  "/opt/pot/jails/test/m/mnt" (as the grep would be for "/opt/pot/jails/test/m/mnt "). Probably a corner case, but it still would be nice if such things could be avoided programmatically.

Reply:
`$_proot/$_mnt_p` is the second element in the file, it's the mountpoint inside the jail (the destination).
The same source could be mounted multiple times, while you can only have one mountpoint.

About your corner case, yes, you are right, it's pathological, that means valid and possible.
Do you have nay suggestion on how to "easily" tackle it?
The only solutions I have in mind are:
* ask the user if a grep returns multiple entries (not script-friendly, tho)
* match the entire string, from ^ to $. However, in this case, we would need to ask the user to input the source and the options (a bit annoying IMO)

I'm open to any suggestion :)

Reply:
> $_proot/$_mnt_p is the second element in the file, it's the mountpoint inside the jail (the destination).

I thought that this is what the first grep was about, seems like I'm confused.

> I'm open to any suggestion :)

My solution would be to simply not allow whitespace characters in mountpoints and have a regex check enforcing that.